crates/ruff_cli/src/diagnostics.rs

```diff
@@ -110,15 +110,19 @@ pub(crate) fn lint_path(
     // to cache `fixer::Mode::Apply`, since a file either has no fixes, or we'll
     // write the fixes to disk, thus invalidating the cache. But it's a bit hard
     // to reason about. We need to come up with a better solution here.)
-    let metadata = if cache.into() && noqa.into() && autofix.is_generate() {
-        let metadata = path.metadata()?;
-        if let Some((messages, imports)) = cache::get(path, package, &metadata, settings) {
-            debug!("Cache hit for: {}", path.display());
-            return Ok(Diagnostics::new(messages, imports));
+    let caching = match package_cache {
+        Some(package_cache) if noqa.into() && autofix.is_generate() => {
+            let relative_path = package_cache
+                .relative_path(path)
+                .expect("wrong package cache for file");
+            let last_modified = path.metadata()?.modified()?;
```

Comment:
Is this identical to `FileTime::from_last_modification_time`? I can't remember why, but I feel like I had to change our choice of timestamp at least once, so it's worth double-checking what we used before and why, and if/how this differs.

Reply:
`FileTime::from_last_modification_time` uses `mtime`
(https://github.com/alexcrichton/filetime/blob/05d09524fc4939b18851ddbec721b64ea1188d15/src/unix/mod.rs#L73-L78) and so does `Metadata::modified` per the docs (
https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.modified).